server/db/models/users.js

```diff
@@ -0,0 +1,258 @@
+//define Users model here
+const Sequelize = require('sequelize');
+const { db } = require('./../database.js');
+
+const { UUID, UUIDV4, STRING, BOOLEAN, INTEGER, DATEONLY } = Sequelize;
+
+const Users = db.define('users', {
+    userId: {
+        primaryKey: true,
+        type: UUID,
+        defaultValue: UUIDV4,
+    },
+    userType: {
+        type: STRING,
+        allowNull: false,
+        //may need to update this later depends on how a user chooses his type
+        validate: {
+            isIn: [
+                ['Guest', 'Existing customer', 'Admin']
+            ],
+        },
+    },
+    loggedIn: {
+        type: BOOLEAN,
+    },
+    email: {
+        type: STRING,
+        allowNull: false,
+        unique: true,
+        validate: {
+            notEmpty: {
+                args: true,
+                msg: 'Email address cannot be empty',
+            },
+            notNull: {
+                arg: true,
+                msg: 'Email address cannot be null',
+            },
+            isEmail: {
+                arg: true,
+                msg: 'Email address must be valid',
+            }
+        },
+    },
+    password: {
```

Comment:
I'm not positive how a guest to the site would even be put into the database unless they were checking out. They also wouldn't have an email in the database which can't be null. Will probably have to adjust, possibly separate guests from users/admins

Reply:
Good point @ShelOrock. What if, for simplicity, we handle email and password validation on the client for now. Then I suppose we'll need to add guests to the users database before checkout with null email address, username, and password. That way every order will have a customer record. I'm open to other solutions though. Lmk what you think!

Reply:
I customized the email and password validation. Now both of them cannot be null if user type is not guest.